rfcs/rfc00XX.md

```diff
@@ -0,0 +1,257 @@
+    39:        $opt_for{description} = $description;
+    DB<2> x status()
+    Undefined subroutine &main::status called at ...
+
+The subroutine is bound a compile-time and removed from the namespace. You can
+step into it, but you can't call it directly from the debugger. This should be
+considered.
+
+* Exporting Spec
+
+We often have the following near the top of our code:
+
+    our @EXPORT    = qw(quux);
+    our @EXPORT_OK = qw(
+        bar
+        baz
+        bay
+    )
+    our %EXPORT_TAGS = qw(
+        'all'     => \@EXPORT_OK,
+        'strings' => qw(bay baz),
+    );
+
+Note that there is a need to duplicate some function names.
+
+Some developers like a single exporting spec at the top of a module. Others
+like viewing a function and seeing directly if it can be exported:
+
+    sub foo :export () {
+        ...
+    }
+
+* The `import()` method
+
+This one is tricky because we have to maintain backwards-compatibility.
+
+If there is no `import()` method defined in the class (inheriting does not
+count?), then any attempt to import a function not designated for export should
+be a fatal error:
+
+    $ perl -MMy::Module=whee -E 1
+    "whee" is not exported by the My::Module module
+    Can't continue after import errors at -e line 0.
+    BEGIN failed--compilation aborted.
+
+For the following, we'll use the following stub example of an `import()`
+method:
+
+    package My::Module {
+        use v5.38;
+
+        sub import ($class, @args) {
+            ...
+        }
+        sub bar  :export            {...} # @EXPORT_OK
+        sub baz  :export(:strings)  {...} # %EXPORT_TAGS
+        sub bay  :export(:strings)  {...} # %EXPORT_TAGS
+        sub quux :export(:ALWAYS)   {...} # @EXPORT
+        sub whee                    {...} # not exported
+    }
+
+If an `import()` method is defined, it's possible that it might have special
+importing functionality provided, but we will _still_ allowing importing of
+functions with `:export` tags. We pass the import arguments unchanged to the
+`import()` method (in `@args`, above) and the module author will be
+responsible for filtering those manually. This seems unfortunate, but if
+someone uses `use v5XX`, their import mechanism shouldn't suddenly break.
+
+As a counter-argument, it's unlikely someone will be using `:export`
+attributes, so we could simply filter out all subnames so they don't populate
+`@args`, or filter out all subnames tagged with `:export`.
+
+For the above, that means that these are problematic:
+
+    use My::Module ':srtings';
+    use My::Module 'whee';
+
+For `:srtings`, that's almost certainly misspelled. For `whee`, that's not
+supposed to be exported, but perhaps the module author wants to allow that in
+the argument list.
+
+To side-step this, we could create an alternate to the `import()` method,
+perhaps as something analogous to a phaser, but taking arguments, or a new
+`IMPORT` method. Bike-shedding may now commence!
```

Comment:
I admit I am somewhat swayed by the idea of an `IMPORT` phaser, see below.

Reply:
I don't see the advantage of a phaser over just a method, as is currently done. A phaser accepting arguments is rather strange to start with. And then it needs some way to process some of them and allow others to be processed by other phasers. None of this is an issue for a method.